syntax = "proto3";

package rpc;

enum MessageType {
  RPC_TYPE_UNKNOW = 0;
  RPC_TYPE_REQUEST = 1;
  RPC_TYPE_RESPONSE = 2;
  RPC_TYPE_ERROR = 3;
}

message RpcMessage {
  uint32 id = 1;
  MessageType type = 2;
  string service_name = 3;
  string method_name = 4;

  bytes request = 5;
  bytes response = 6;
}